### PR TITLE
ENH old smithy not compatible with new conda-build

### PR DESCRIPTION
**Pull request by beckermr**
Checklist

* [ ] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md)).
* [ ] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [ ] Ran `pre-commit run -a` and ensured all files pass the linting checks.
* [ ] Ran `python show_diff.py` and posted the output as part of the PR.
* [ ] Modifications won't affect packages built in the future. 

xref: https://github.com/conda-forge/conda-smithy/pull/1816

### CODE REVIEW

File: recipe/patch_yaml/conda-smithy.yaml [hunk 1]
```diff
@@ -19,3 +19,11 @@ then:
   - tighten_depends:
       name: conda
       upper_bound: "5"
+if:
+  name: conda-smithy
+  version: 3.30.1
+then:
+  - replace_depends:
+    old: conda-build?( *)
+    new: conda-build<3.28.2
```
Comment (mbargull) [inline recipe/patch_yaml/conda-smithy.yaml:28]: This would remove the lower bounds, right?

File: recipe/patch_yaml/conda-smithy.yaml [hunk 2]
```diff
@@ -19,3 +19,11 @@ then:
   - tighten_depends:
       name: conda
       upper_bound: "5"
+if:
```
Comment (beckermr) [inline recipe/patch_yaml/conda-smithy.yaml:22]:
```suggestion
---
if:
```

File: recipe/patch_yaml/conda-smithy.yaml (hunk 1 above)
Comment (beckermr) [inline recipe/patch_yaml/conda-smithy.yaml:28]:
```suggestion
  - tighten_depends:
    name: conda-build
    upper_bound: 3.28.2
```

Comment (beckermr) [reply recipe/patch_yaml/conda-smithy.yaml]: Good catch. Should be fixed now.

**Comment by mbargull** [comment]
xref: https://github.com/conda/conda-build/issues/5125

File: recipe/patch_yaml/conda-smithy.yaml [hunk 3]
```diff
@@ -19,3 +19,11 @@ then:
...
       name: conda
       upper_bound: "5"
+---
+if:
+  name: conda-smithy
+  version: 3.30.1
+then:
+  - tighten_depends:
+    name: conda-build
+    upper_bound: 3.28.2
```
Comment (beckermr) [inline recipe/patch_yaml/conda-smithy.yaml:29]:
```suggestion
      name: conda-build
      upper_bound: 3.28.2
```

**Comment by beckermr** [comment]
diff:

```diff
================================================================================
================================================================================
noarch
noarch::conda-smithy-3.30.1-pyhd8ed1ab_0.conda
-    "conda-build >=3.18.3",
+    "conda-build >=3.18.3,<3.28.2.0a0",
```

**Review by mbargull — APPROVED** [review]

**Comment by beckermr** [comment]
Well we should patch all copies of smithy.

**Comment by jakirkham** [comment]
Does that mean this patch is not sufficient? If so, should we raise an issue to track?

**Comment by mbargull** [comment]
> Does that mean this patch is not sufficient? If so, should we raise an issue to track?

https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/622

**Comment by jakirkham** [comment]
Ah ok. Thanks Marcel! 🙏

**Pull request merged**